simulation/chart.py

```diff
@@ -1,232 +1,427 @@
+    "c",  # cyan
+    "m",  # magenta
+    "y",  # yellow
+    "k",  # key (black)
+    "w",  # white
+]
 
-def create_chart(
-    layout: pg.GraphicsLayoutWidget,
-    title: Optional[str] = None,
-    height: Optional[int] = None,
-    legend_width: Optional[int] = None,
-    x_label: Optional[str] = None,
-    x_range: Optional[Tuple[float, float]] = None,
-    x_log: Optional[bool] = None,
-    y_label: Optional[str] = None,
-    y_range: Optional[Tuple[float, float]] = None,
-    y_log: Optional[bool] = None,
-) -> pg.PlotItem:
-    """Add a chart with legend for plots as a row
 
-    Args:
-        layout(GraphicsLayoutWidget): Parent layout
-        title (str, optional): Title to display on the top. Defaults to None.
-        height (int, optional): Prefred height. Defaults to None.
-        legend_width (int, optional):Legend width. Defaults to None.
-        x_label (str, optional): Label text to display under x axis. Defaults to None.
-        x_range (Tuple[float,float], optional): Constant range(min,max value) of x axis. Defaults to None.
-        x_log (bool, optional): Use logarithmic scale to display x axis. Defaults to None.
-        y_label (str, optional): Label text to display next to y axis. Defaults to None.
-        y_range (Tuple[float,float], optional): Constant range(min,max value) of y axis. Defaults to None.
-        y_log (bool, optional): Use logarithmic scale for y axis. Defaults to None.
-    Returns:
-        PlotItem: a chart to contain plots
+class IndexedColor(NamedTuple):
+    """Color from a single index. Useful for stepping through a predefined list of colors.
+
+    Attributes:
+        index (int): predefined color index
+        hues (int): number of steps to generate
     """
-    chart = layout.addPlot(title=title)
-    if height is not None:
-        chart.setPreferredHeight(300)
-    legend = chart.addLegend(offset=None)
-    vb: pg.ViewBox = layout.addViewBox()
-    if legend_width is not None:
-        vb.setFixedWidth(legend_width)
-    legend.setParentItem(vb)
-    legend.anchor((0, 0), (0, 0))
-    chart.setClipToView(True)
 
-    if x_label is not None:
-        chart.setLabel("bottom", x_label)
-    if x_log is not None:
-        chart.setLogMode(x=x_log)
-    if x_range is not None:
-        chart.setXRange(*x_range)
+    index: int
+    hues: int
 
-    if y_label is not None:
-        chart.setLabel("left", y_label)
-    if y_log is not None:
-        chart.setLogMode(y=y_log)
-    if y_range is not None:
-        chart.setYRange(*y_range)
 
-    layout.nextRow()
+Color = Union[RGBAColor, NamedColor, IndexedColor, int, float, str]
+"""Color based on model, predefined values, single greyscale value (0.0 - 1.0) , color index or string representing color (“#RGB”,“#RGBA”,“#RRGGBB”,“#RRGGBBAA”)."""
 
-    return chart
 
+@singledispatch
+def _make_color(color: RGBAColor):  # pyright:ignore [reportGeneralTypeIssues]
+    """Convert RGBA color value to QColor"""
+    if color.A is None:
```

Comment:
If you use a reasonable default for `A`, we can remove this `None` RTTI.

Reply:
https://github.com/semiotic-ai/autoagora-agents/pull/21/commits/0d488125217f7472c649d43cfd26daae40c6cbec